### ISSUE TITLE
ubuntu 22 jammy - few issues with sunshine

### ISSUE THREAD

**Issue by Jonathan34** [stefanjenkner/ec2-gaming-sunshine#56]
Hi - thanks for putting this together.
I was able to create the instance with spot/on-demand.

I'm having few issues running sunshine though. and the setup instructions do not help as few bits are missing. hoping the below will help others.

As i understand, you create a sunshine user to run sunshine vs running under ubuntu.

1. After the initial setup, the apt is broken as steam installs the apt repo as well. `rm /etc/apt/sources.list.d/steam.list` does the trick.

2. sunshine has no shell so cannot access dbus for most of the commands `su -c "systemctl status sunshine" sunshine`.
one need to exec a shell for the user sunshine once to properly setup the profile:

```
ubuntu$id sunshine
  uid=1000(sunshine) gid=1000(sunshine) groups=1000(sunshine),29(audio),44(video),46(plugdev),100(users),107(input),119(netdev)
ubuntu$ sudo machinectl shell --uid=1000
sunshine$ systemctl status sunshine
```

3. although sunshine is part of input group, it still cannot get permission to access input.
`sunshine : sunshine audio video plugdev users input netdev`

full log:

```
sunshine@i-xxx:~$ cat .config/sunshine/sunshine.log
[2023:08:15:20:25:11]: Info: Sunshine version: v0.20.0
[2023:08:15:20:25:11]: Warning: Failed to create system tray
[2023:08:15:20:25:11]: Error: Couldn't load cuda: -1
[2023:08:15:20:25:11]: Warning: Couldn't find /dev/dri, kmsgrab won't be enabled
[2023:08:15:20:25:11]: Error: Unable to initialize capture method
[2023:08:15:20:25:11]: Error: Platform failed to initialize
[2023:08:15:20:25:11]: Error: Could not create Sunshine Mouse: Permission denied
[2023:08:15:20:25:11]: Error: Could not create Sunshine Touchscreen: Permission denied
[2023:08:15:20:25:11]: Error: Could not create Sunshine Keyboard: Permission denied
[2023:08:15:20:25:11]: Error: Unable to create some input devices! Are you a member of the 'input' group?
[2023:08:15:20:25:11]: Info: Falling back to XTest
[2023:08:15:20:25:11]: Info: // Testing for available encoders, this may generate errors. You can safely ignore those errors. //
[2023:08:15:20:25:11]: Info: Trying encoder [nvenc]
[2023:08:15:20:25:12]: Info: Encoder [nvenc] failed
[2023:08:15:20:25:12]: Info: Trying encoder [vaapi]
[2023:08:15:20:25:14]: Info: Encoder [vaapi] failed
[2023:08:15:20:25:14]: Info: Trying encoder [software]
[2023:08:15:20:25:15]: Info: Encoder [software] failed
[2023:08:15:20:25:15]: Fatal: Couldn't find any working encoder
[2023:08:15:20:25:15]: Error: Video failed to find working encoder
[2023:08:15:20:25:15]: Info: Open the Web UI to set your new username and password and getting started
[2023:08:15:20:25:15]: Info: File /home/sunshine/.config/sunshine/sunshine_state.json doesn't exist
[2023:08:15:20:25:15]: Error: Failed to create client: Daemon not running
[2023:08:15:20:25:15]: Info: Configuration UI available at [https://localhost:47990]
[2023:08:15:20:26:11]: Info: Terminate handler called
```

`sudo setcap -r $(readlink -f $(which sunshine))` seems to help with a reboot of the sunshine service.

I got it to work ultimately but the installation remains a bit flaky at this time.

**Comment by stefanjenkner** [comment]
Thanks a lot for detailed feedback and the proposed solutions, @Jonathan34!

The root cause seems to be the broken APT which I fixed now with #61. I hope this works for you.

Unfortunately, I cannot reproduce the issues with sunshine v0.18.4 (which is installed automatically) but I'll come up with a separate PR to update sunshine to the recent version (I noticed you tested with v0.20.0) and for now I'll stick to v0.18.4. ✌️

**Comment by Jonathan34** [comment]
Thanks! your repo saved me quite some time to setup resources. so thanks for sharing it.

I also added support for DDNS and dynamic ip update with no-ip.com (so you do not need an elastic ip) and automatically installed packages like firefox, lutrix, wine, wine-tricks for example.

sunshine 0.20 has some sysctl startup service issue as well, it now depends on xdg but it never really starts. outside of that, it works fine.
